logp/ratelimited.go

```diff
@@ -0,0 +1,96 @@
+package logp
+
+import (
+	"sync"
+	"sync/atomic"
+	"time"
+)
+
+// RateLimitedLogger is a logger that limits log messages to once within a
+// specified period.
+// It is intended for logging events that occur frequently, providing a summary
```

Comment:
Why not use https://pkg.go.dev/golang.org/x/time/rate or even https://github.com/elastic/beats/tree/main/libbeat/processors/ratelimit

Reply:
There is a custom implementation of the token bucket algorithm in https://github.com/elastic/beats/blob/main/libbeat/processors/ratelimit/token_bucket.go but I'd honestly prefer we replace that the one in x/time/rate as well.

Reply:
well apart from the fact I missed there is a `[Every](https://pkg.go.dev/golang.org/x/time/rate#Every)`, the work to accumulate the number of events to then log the summary would be the same. Also using the ticker it's a rather straight forward implementation. Thus I don't find it necessary to use https://pkg.go.dev/golang.org/x/time/rate